### ISSUE TITLE
Editing of comments improvements (element start does not work, closing tag)

### ISSUE THREAD

**Issue by WolfgangHG** [eclipse/lemminx#1154]
This happens with the "Generic Text Editor" of Eclipse 2021-12.

**First problem:**
Commenting xml snippets is a bit strange: if you want to start a comment and the cursor is placed on the opening "<" of an xml element, and then you type "<", I would expect this char to be entered. But it seems to overwrite the opening "<" of the next element, and if you continue typing "!--", strange things will happen: 

`<!--scope>provided</--!scope>`

My expected result is:

`<!--<scope>provided</scope>`

...so that I can enter the closing comment tag at the end of the line.

So you have to take care that there is always a space after the current position before you start entering a comment.

**Second problem:**
When typing "<" and there is whitespace afterwards, there is always a ">" entered, also. This is OK if you pick the "`<!--`"  suggestion from the popup and press enter - then the result is "`<!-- -->`".
But if you just close/ignore the popup because you want to comment an existing block and type "`-->`" at e.g. the end of the line, you have to delete the ">" always. Would it be possible to remove the autogenerated ">" automatically if a "!" is typed after the "<" or if the current line contains further text?

**Comment by angelozerr** [comment]
@WolfgangHG could you do 2 little demos with your problem. I'm sorry I don't understand very well your usecases. Thanks.

**Comment by WolfgangHG** [comment]
OK, I will try.

**First problem:**
This snippet is from a pom.xml. I have a dependency with a "version" element, and because I want to do some testing with the version, I want to comment the "version" line. So set the cursor just before the opening tag:
![problem1_step1](https://user-images.githubusercontent.com/138049/152044955-baec744e-ac5c-48ca-ad59-cf5034a099f4.png)

Press the `<` key to start the comment.
Instead of entering a new `<` character, it seems to overwrite the existing tag:
![problem1_step2](https://user-images.githubusercontent.com/138049/152045079-39409904-45f2-4761-9fe6-cf4c984bf443.png)

If you don't notice this and continue typing the rest of the comment (`!--`), the whole line seems to be broken, as the comment part is added also to the closing tag:
![problem1_step3](https://user-images.githubusercontent.com/138049/152045435-09d69bc4-dfbc-4446-a655-5fc82a258579.png)

I would expect that the `<`  is not ignored, but just entered in the file "as is".

It works if there is a space beetween comment and next opening tag. But in sitatutions where I want to comment just one line, I am used to "go to the xml start and type the comment". Some tools cause trouble, because they automatically insert the closing comment tag, but I am used to this behaviour and delete the comment. But the trouble with the opening tag is new to me.

**Comment by WolfgangHG** [comment]
**Second problem/question**
Start in an empty line and type `<`. Automatically, a `>` is entered:
![problem2_step1](https://user-images.githubusercontent.com/138049/152046506-5865174b-078e-46b1-a7b1-a4c07103908b.png)

Type the rest of the comment (`!--`). Don't press enter:
![problem2_step2](https://user-images.githubusercontent.com/138049/152046651-396f9348-3346-4600-95ea-fc42ac8cafb6.png)

Just close the code completion popup, as I want to enter the closing comment tag at another line (see caret position in screenshot): The closing `>` remains and I have to delete it myself.
![problem2_step3](https://user-images.githubusercontent.com/138049/152046947-24e5c637-9345-46ef-a2f0-9b91f7137b3c.png)

If  I press enter in the code completion popup, the comment closing tag is entered - that is OK for me, though I have to delete it myself in my use case. At least the XML is valid:
![problem2_step4](https://user-images.githubusercontent.com/138049/152047179-b456c2c4-1147-4b62-8bf0-6c5ebf60931f.png)

So, I think it would be reasonable to remove the auto-inserted `>` in case the code completion popup is not committed by pressing enter, but clicking somewhere else. Or even don't add it in case of comments?

**Comment by angelozerr** [comment]
>Second problem/question
>Start in an empty line and type <. Automatically, a > is entered:

I agree with you, it's an anoying behavior and that's why we removed this auto close pairs in vscode-xml. IMHO WWD should remove that, but since it doesn't manage auto close tag, I think it's the reason why '>' is generated.

@mickaelistria I think WWD should remove https://github.com/eclipse/wildwebdeveloper/blob/059aedf64486323cd515e8bbdbf6f8747644b3a4/org.eclipse.wildwebdeveloper.xml/language-configurations/xml/xml.language-configuration.json#L10

@WolfgangHG you can disable auto pairs in WWD like this:

![image](https://user-images.githubusercontent.com/1932211/152111924-07bf9bb6-22df-4373-9917-ee8ec6effd9e.png)

**Comment by angelozerr** [comment]
> First problem:
This snippet is from a pom.xml. I have a dependency with a "version" element, and because I want to do some testing with the version, I want to comment the "version" line. So set the cursor just before the opening tag:

It's again an auto close pairs. If you disable it, you will not have this problem.

@mickaelistria it's an another usecase which shows that auto pair for < is a bad idea. In vscode-xml we have not that.

In other words this issue is an issue for WWD, not for LemMinX.

**Comment by WolfgangHG** [comment]
Yes, without this setting, XML editing works much better.  @angelozerr If this issue shall be moved to another project, could you handle this? I don't have enough knowledge of the internals of your projects, and it seems I don't have the privileges to move issues, anyway.

**Comment by angelozerr** [comment]
@WolfgangHG I created a PR at https://github.com/eclipse/wildwebdeveloper/pull/784

Let's continue the discussion in this PR.
